### PR TITLE
[JBPM-9129] Start process without triggering a start node

### PR DESCRIPTION
**Pull request by elguardian**
Adding interfaces to support restart process

### CODE REVIEW

**Comment by elguardian** [comment]
Jenkins retest this please

**Comment by gmunozfe** [comment]
jenkins retest this

File: kie-api/src/main/java/org/kie/api/definition/process/Node.java [hunk 1]
```diff
@@ -31,6 +31,17 @@
      */
     long getId();
 
+    /**
+     * The node id provided in the bpmn 
+     */
+    String getUniqueId();
```
Comment (krisv) [inline kie-api/src/main/java/org/kie/api/definition/process/Node.java:37]: This is probably going to confuse users, as it's two similar fields with similar description even.  getUniqueId() is an existing operation in the internal Node interface already, that simply concatenates the nodeId with the nodeIds of the parent.  This is somewhat outdated, as we later switched how node ids (long) are generated so they are unique across a process, so I think this operation is outdated.
I would recommend to keep getUniqueId() unchanged (for backwards compatibility) but keep it in internal node interface only, not promote it to kie-api.  That way we would only have getNodeUniqueId() here, wdyt?

Comment (elguardian) [reply kie-api/src/main/java/org/kie/api/definition/process/Node.java:37]: fine.

**Comment by elguardian** [comment]
done @krisv

File: kie-api/src/build/revapi-config.json [hunk 2]
```diff
@@ -1,41 +1,85 @@
...
+                        "regex": true,
+                        "exclude": []
+                    }
+                }
+            }
+        }
+    },
+    "ignores": {
+        "revapi": {
+            "_comment": "Changes between 7.33.0.Final and the current branch. These changes are desired and thus ignored.",
```
Comment (afalhambra) [inline kie-api/src/build/revapi-config.json:16]: Change the comment it to something more meaningful as this from previous file.

File: kie-api/src/build/revapi-config.json [hunk 3]
```diff
@@ -1,41 +1,85 @@
...
+                },
+                {
+                    "code": "java.method.addedToInterface",
+                    "new": "method java.lang.String org.kie.api.definition.process.Node::getUniqueId()",
+                    "package": "org.kie.api.definition.process",
+                    "classSimpleName": "Node",
+                    "methodName": "getUniqueId",
+                    "elementKind": "method",
+                    "justification":" get the the unique id"
+                },
```
Comment (afalhambra) [inline kie-api/src/build/revapi-config.json:71]: +1

File: kie-api/src/main/java/org/kie/api/runtime/process/ProcessRuntime.java [hunk 4]
```diff
@@ -48,6 +48,15 @@
     ProcessInstance startProcess(String processId,
                                  Map<String, Object> parameters);
 
+    /**
+     * Starts a new process instances starting from the nodes given
+     * @param processId the id of the process that should be started
+     * @param params the process variables that should be set when starting the process instance
+     * @param nodeInstancesIds the node instances id that should be triggered
```
Comment (afalhambra) [inline kie-api/src/main/java/org/kie/api/runtime/process/ProcessRuntime.java:55]: Please change javadoc. 

```suggestion
     * @param nodeId the node instances id that should be triggered
```

File: kie-api/src/main/java/org/kie/api/runtime/manager/audit/NodeInstanceLog.java [hunk 5]
```diff
@@ -37,10 +37,27 @@
...
-     * Inidcates that the node instace was left because an abort operation (it is not active anymore)
+     * Indicates that the node instance was left because an abort operation (it is not active anymore)
      */
     public static final int TYPE_ABORTED = 2;
 
+    /**
+     * Indicates that the node was timed and cancelled
+     */
+
+    public static final int TYPE_TIMED = 3;
```
Comment (gmunozfe) [inline kie-api/src/main/java/org/kie/api/runtime/manager/audit/NodeInstanceLog.java:48]: rename to TYPE_SKIPPED

File: kie-api/src/build/revapi-config.json [hunk 6]
```diff
@@ -1,41 +1,85 @@
...
+                {
+                    "code": "java.method.addedToInterface",
+                    "new": "method java.lang.Object org.kie.api.runtime.manager.audit.VariableInstanceLog::getValueObject()",
+                    "package": "org.kie.api.runtime.manager.audit",
+                    "classSimpleName": "VariableInstanceLog",
+                    "methodName": "getValueObject",
+                    "elementKind": "method",
+                    "justification": "necessary when full audit is neede to restore the original object"
+                },
```
Comment (afalhambra) [inline kie-api/src/build/revapi-config.json:44]: Not needed. No such method in the class

File: kie-api/src/build/revapi-config.json [hunk 7]
```diff
@@ -1,41 +1,85 @@
...
+                {
+                    "code": "java.method.addedToInterface",
+                    "new": "method org.kie.api.runtime.process.ProcessInstance org.kie.api.runtime.process.ProcessRuntime::startProcessFromNodeIds(java.lang.String, java.util.Map<java.lang.String, java.lang.Object>, java.lang.String[])",
+                    "package": "org.kie.api.runtime.process",
+                    "classSimpleName": "ProcessRuntime",
+                    "methodName": "startProcessFromNodeIds",
+                    "elementKind": "method",
+                    "justification": "new functionality added"
```
Comment (afalhambra) [inline kie-api/src/build/revapi-config.json:52]: Can you please specify what this functionality is added?

File: kie-internal/src/main/java/org/kie/internal/process/CorrelationAwareProcessRuntime.java [hunk 8]
```diff
@@ -56,6 +56,20 @@ ProcessInstance startProcess(String processId, CorrelationKey correlationKey,
...
+     * Starts a process with the values supplied from the current nodes
+     * 
+     * @param deploymentId deployment information for the process's kjar
+     * @param processId The process's identifier
+     * @param correlationKey correlation key to be assigned to process instance - must be unique 
+     * @param params process variables
+     * @return process instance identifier
+     * @throws RuntimeException in case of encountered errors
+     * @throws DeploymentNotFoundException in case deployment with given deployment id does not exist
+     * @throws DeploymentNotActiveException in case deployment with given deployment id is not active
```
Comment (afalhambra) [inline kie-internal/src/main/java/org/kie/internal/process/CorrelationAwareProcessRuntime.java:69]: Please correct javadocs. They don't match with what the method signature.

File: kie-api/src/build/revapi-config.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (elguardian) [reply kie-api/src/build/revapi-config.json:71]: done

File: kie-api/src/main/java/org/kie/api/runtime/manager/audit/NodeInstanceLog.java (hunk 5 above)
Comment (elguardian) [reply kie-api/src/main/java/org/kie/api/runtime/manager/audit/NodeInstanceLog.java:48]: missed to push this change.

File: kie-api/src/build/revapi-config.json (hunk 3 above)
Comment (elguardian) [reply kie-api/src/build/revapi-config.json:71]: done

File: kie-api/src/main/java/org/kie/api/runtime/process/ProcessRuntime.java (hunk 4 above)
Comment (elguardian) [reply kie-api/src/main/java/org/kie/api/runtime/process/ProcessRuntime.java:55]: done

File: kie-api/src/build/revapi-config.json (hunk 7 above)
Comment (elguardian) [reply kie-api/src/build/revapi-config.json:52]: done

File: kie-internal/src/main/java/org/kie/internal/process/CorrelationAwareProcessRuntime.java (hunk 8 above)
Comment (elguardian) [reply kie-internal/src/main/java/org/kie/internal/process/CorrelationAwareProcessRuntime.java:69]: done

**Comment by afalhambra** [comment]
Jenkins retest this please

**Pull request merged**